### PR TITLE
Ballistic Supremacy - The way of the WT

### PR DESCRIPTION
**Pull request by MajManatee**
# Document the changes in your pull request

I lie awake at night. The horrors of this world fill my mind.
Felinids...
Greytiders...
Assistants...

**Lizards...**

I snap back to consciousness, drenched in sweat. Not out of fear, but out of rage.
I grab my rifle from beside my bed, and rack it with malicious intent.

_I will kill them all._

 
 
 
Anyways, the voices have been especially loud lately so I put together this PR. Basically you get three new mag types for the WT, Researchable after exotic ammo and ballistic tech. This PR also moves Techshells for shotguns to the new node, called "Experimental Ammunition".

Anyways balance council please @ me on discord with suggestions otherwise I wont see them. thanks

# Spriting

Reskins of magazines. 

Venom
image

Kraken
image

Airburst
image

# Wiki Documentation

Venom Mag - 20 Rounds, 6 damage per bullet, but injects 4u toxin into its target and does 6 toxin damage for a lower death.
Kraken Mag - 20 Rounds, -60 AP, 25 dmg per round, -30 wound chance
Airbust Mag - 10 Rounds, 3 dmg bullet till it travels around 3 tiles, then it splits into 3 pellets that spread out, and do 10 dmg each.

New techweb node - "Experimental Ammunition" - Contains the 3 mags, and tech shells, which were previously in exotic ammo. Requires Exotic ammo and ballistic tech to be researched.

# Changelog

:cl:  
rscadd: WT Venom mag - Dealing toxin damage through microneedles or something 
rscadd: WT Kraken mag - Super reinforced flat-tipped bullets that rip through flesh, but are canceled by any armor
rscadd: WT Airburst mag - Fires a round that, after traveling a short distance, pops and splits into three smaller rounds. 
tweak: Tech shells moved to the new "Experimental Ammo" node
/:cl:

### CODE REVIEW

File: code/modules/projectiles/projectile/bullets/smg.dm [hunk 1]
```diff
@@ -50,3 +50,22 @@
...
+		var/mob/living/carbon/victim = target
+		victim.reagents.add_reagent(/datum/reagent/toxin, 6)
+		victim.adjustToxLoss(6)
+
+	return ..()
+/obj/item/projectile/bullet/c46x30mm/kraken
+	name = "4.6x30mm kraken bullet"
+	damage = 22
+	armour_penetration = -50
+	sharpness = SHARP_EDGED
```
Comment (Moltijoe) [inline code/modules/projectiles/projectile/bullets/smg.dm]: not SHARP_POINTY?

File: code/modules/projectiles/projectile/bullets/smg.dm [hunk 2]
```diff
@@ -50,3 +50,22 @@
...
 	stamina = 22
+
+/obj/item/projectile/bullet/c46x30mm/venom
+	name = "4.6x30mm venom bullet" // Doesnt actually inject toxin. Just does toxin damage.
+	damage = 6
+
+/obj/item/projectile/bullet/c46x30mm/venom/on_hit(atom/target, blocked)
+	if((blocked != 100) && iscarbon(target))
+		var/mob/living/carbon/victim = target
+		victim.reagents.add_reagent(/datum/reagent/toxin, 6)
```
Comment (Moltijoe) [inline code/modules/projectiles/projectile/bullets/smg.dm]: this is actually just injecting toxin though?

**Review by Moltijoe — CHANGES_REQUESTED** [review]
please update wiki documentation to align with what's actually in the code

**Comment by Mqiib** [comment]
Fix your descriptions in the pr and in code, make the magazines not called rubber bullet magazines, and reduce the injected toxin to 4 units. Also IMO the kraken rounds should be blunt rather than sharp_edged because otherwise they're literally just printable HP rounds.

File: code/modules/projectiles/projectile/bullets/smg.dm (hunk 2 above)
Comment (MajManatee) [reply code/modules/projectiles/projectile/bullets/smg.dm]: i forgot to update the documentation 💀

**Comment by MajManatee** [comment]
Documentatnion updated, Changes made, nerfed Kraken more against armor because idk its funnier this way.

**Review by cuackles — APPROVED** [review]
put the sprites on the description or perish

**Review by Moltijoe — CHANGES_REQUESTED** [review]
please mention that you've also included airburst rounds

**Comment by MajManatee** [comment]
made a draft so that molti stops yelling at me for chaning scope constnatly. 

ILL UPDATE THE DOCUMENTATION WHEN ITS DONE. till then yall can suck an egg.

File: code/modules/projectiles/projectile/bullets/smg.dm [hunk 3]
```diff
@@ -50,3 +50,34 @@
...
+	name = "4.6x30mm airburst pellet"
+	damage = 10
+/obj/item/projectile/bullet/c46x30mm/airburst
+	name = "4.6x30mm airburst bullet"
+	damage = 2 // its just a casing
+	range = 6
+
+/obj/item/projectile/bullet/c46x30mm/airburst/on_range(mob/user)
+	var/obj/item/ammo_casing/c46x30mm/airburst_pellet/P = new(get_turf(src))
+	P.fire_casing()
```
Comment (Moltijoe) [inline code/modules/projectiles/projectile/bullets/smg.dm]: you requested my review so here it is
this won't work, it needs parameters

**Review by Moltijoe — DISMISSED** [review]

File: code/modules/projectiles/projectile/bullets/smg.dm (hunk 3 above)
Comment (MajManatee) [reply code/modules/projectiles/projectile/bullets/smg.dm]: i didnt mean to, i am well aware it doesnt work

File: code/modules/projectiles/projectile/bullets/smg.dm [hunk 4]
```diff
@@ -50,3 +50,34 @@
...
+	damage = 10
+/obj/item/projectile/bullet/c46x30mm/airburst
+	name = "4.6x30mm airburst bullet"
+	damage = 2 // its just a casing
+	range = 6
+
+/obj/item/projectile/bullet/c46x30mm/airburst/on_range(obj/item/projectile/bullet/c46x30mm/airburst/A, mob/living/carbon/human/D)
+	var/obj/item/ammo_casing/c46x30mm/airburst_pellet/P = new(get_turf(src))
+	P.fire_casing(D, A)
+	..()
```
Comment (Moltijoe) [inline code/modules/projectiles/projectile/bullets/smg.dm]: have you tested this? does it work? because i have no reason to believe it will

Comment (MajManatee) [reply code/modules/projectiles/projectile/bullets/smg.dm]: you did it in YOUR pr

Comment (Moltijoe) [reply code/modules/projectiles/projectile/bullets/smg.dm]: but i was given those variables from elsewhere in the code, on_range isn't passed those parameters

**Comment by Moltijoe** [comment]
there, i made it function to the best i'm willing to spend time on, you'll need someone else to approve the PR now though

**Comment by Moltijoe** [comment]
how come venom rounds inject toxin instead of venom?
the round don't act at all like other venoms rounds

**Comment by MajManatee** [comment]
yeah give sec venom, a notoriously traitor and blatantly powerful killing chem. Great idea

because that's dumb. Its just a name.

**Comment by 00ze-cyclone** [comment]
> yeah give sec venom, a notoriously traitor and blatantly powerful killing chem. Great idea
> 
> because that's dumb. Its just a name.

Name it toxin round so nobody can complain

**Review by Moltijoe — DISMISSED** [review]
please rename the venom rounds so they aren't confused for other venom rounds that inject the venom chemical

**Comment by MajManatee** [comment]
Toxin round is the most boring thing ever
Nobody is confusing it with real venom

**Comment by Moltijoe** [comment]
> Toxin round is the most boring thing ever Nobody is confusing it with real venom

trust me, they WILL
if there's literally any way someone can misunderstand something, there will be someone to misunderstand it

**Comment by MajManatee** [comment]
name changed to be less cool

**Review by ToasterBiome — APPROVED** [review]

**Pull request merged**